gems/aws-sdk-s3/lib/aws-sdk-s3/file_downloader.rb

```diff
@@ -129,6 +140,14 @@ def thread_batches(chunks, param)
                 @params.merge(param.to_sym => chunk)
               )
               write(resp)
+              if @on_checksum_validated &&
+                resp.context[:http_checksum] &&
+                resp.context[:http_checksum][:validated]
```

Comment:
Checking context in the response here is kinda fragile. We also were talking about not having context on responses in the future. Is there a better way to determine this information? Maybe pushing logic to the checksum plugin?

Reply:
This value is set by the checksum plug-in and we decided response context was how we were going to make this information available in V3 when we implemented flexible checksums. 

I guess we could add an additional config option on the client for a similar callback? Any other ideas?

Reply:
Does it make sense to have an API private (ruby public) method on the Checksum plugin that our plugins/implementations can use to check if http checksum is validated?

I guess the alternative is that we have a config option for on_checksum_validated that is called by the checksum middleware - then it wouldn't exist in high level abstraction and be called on each get_object (or part).

Reply:
After more discussion - added a `checksum_validated` method to the `Response` - this makes it easy for users to access (and is now better documented/discoverable as well).